app/main/posts/savedsearches/saved-search-modal.directive.js

```diff
@@ -33,18 +31,9 @@ function SavedSearchModalController($scope, $element, $attrs, $rootScope, $locat
         $scope.searches = [];
         query = query || {};
         SavedSearchEndpoint.query(query).$promise.then(function (searches) {
-            $scope.searches = searches;
-
-            angular.forEach($scope.searches, function (search) {
-                // if this search has a user, and its not the currentUser
-                if (_.isObject(search.user) && search.user.id !== _.result($rootScope.currentUser, 'userId')) {
-                    // Load the user (if we haven't already)
-                    if (!users[search.user.id]) {
-                        users[search.user.id] = UserEndpoint.getFresh({id: search.user.id});
-                    }
-                    // Save user info onto the search itself
-                    search.user = users[search.user.id];
-                }
+            $scope.searches = _.filter(searches, function (search) {
```

Comment:
This looks ok for now. I'm a little concerned it might not scale with lots of searches ie. because we'll get many many results from the DB. 

Thats probably a problem to save later though!

Reply:
Yup, this won't scale. The original approach from the backend was better, or if that was too much, at least adding a filter that supports OR, which I don't think we have.  I tried sending the request with the filters  in the query string, but it didn't work out because it was looking for featured AND my own posts. We can look into it when you get back I think.

Reply:
> The original approach from the backend was better, 
Actually, that's not right :) I have no idea if it was better and it probably wasn't. The option of the OR filter sounds better in retrospective and aligned with the needs in this ticket
@willdoran you helped me in this ticket originally. What do you think? I'll create an issue based on this so we don't forget about it.